src/cmd/run.rs

```diff
@@ -384,29 +407,45 @@ async fn upload_scans<D: Database>(
     let limiter = RateLimiter::direct(quota);
 
     loop {
-        let job = receiver.recv().await.change_context(Error::TaskReceive)?;
-        let meta = ProjectMetadata::new(&job.integration, &job.reference);
-        if limiter.check().is_err() {
-            info!("Integration '{meta}': waiting for rate limit");
-            limiter.until_ready().await;
+        if let Err(err) = execute_upload_scans(ctx, receiver, &limiter).await {
+            warn!("attempt to upload scan failed: {err:#?}");
         }
+    }
+}
 
-        info!("Uploading scan for project: '{meta}'");
-        let locator = fossa::upload_scan(ctx.config.fossa_api(), &meta, &job.cli, job.source_units)
-            .await
-            .change_context(Error::TaskHandle)?;
+#[tracing::instrument(skip_all)]
+async fn execute_upload_scans<D: Database>(
+    ctx: &CmdContext<D>,
+    receiver: &Queue<UploadSourceUnits>,
+    limiter: &RateLimiter<
+        governor::state::NotKeyed,
+        governor::state::InMemoryState,
+        governor::clock::QuantaClock,
+        governor::middleware::NoOpMiddleware<governor::clock::QuantaInstant>,
+    >,
```

Comment:
What a mouthful 😅 

`governor` [is aware of how unwieldy these types are](https://docs.rs/governor/latest/governor/_guide/index.html#type-signatures-for-rate-limiters) and provides a more convenient option:
```suggestion
    limiter: &limiter: &governor::DefaultDirectRateLimiter,
```

This optionally could also be a place where we do a _little_ error handling duplication to make this less unwieldy overall:
```rust
#[tracing::instrument(skip_all)]
async fn upload_scans<D: Database>(
    ctx: &CmdContext<D>,
    receiver: &Queue<UploadSourceUnits>,
) -> Result<(), Error> {
    // This worker is per integration, so the rate limiter should be constructed here instead of globally.
    let quota = Quota::per_minute(nonzero!(1u32));
    let limiter = RateLimiter::direct(quota);

    loop {
        let job = match receiver.recv().await.change_context(Error::TaskReceive) {
            Ok(job) => job,
            Err(err) => {
                warn!("Unable to read enqueued upload job. Details: {err:#?}");
                continue;
            }
        };

        let meta = ProjectMetadata::new(&job.integration, &job.reference);
        if limiter.check().is_err() {
            info!("Integration '{meta}': waiting for rate limit");
            limiter.until_ready().await;
        }

        if let Err(err) = execute_upload_scans(ctx, &meta, job).await {
            warn!("Unable to upload scan for '{meta}': {err:#?}");
        }
    }
}

#[tracing::instrument(skip_all)]
async fn execute_upload_scans<D: Database>(
    ctx: &CmdContext<D>,
    meta: &ProjectMetadata,
    job: UploadSourceUnits,
) -> Result<(), Error> {
    info!("Uploading scan for project: '{meta}'");
    let locator = fossa::upload_scan(ctx.config.fossa_api(), meta, &job.cli, job.source_units)
        .await
        .change_context(Error::TaskHandle)?;

    debug!(scan_id = %job.scan_id, locator = %locator, "Uploaded scan");
    info!("Uploaded scan for project '{meta}' as locator: '{locator}'");

    let remote = job.integration.remote().to_owned();
    let coordinate = job.reference.as_coordinate(&remote);
    let state = job.reference.as_state();

    // Mark this reference as scanned in the local DB.
    ctx.db
        .set_state(&coordinate, state)
        .await
        .change_context(Error::TaskSetState)
}
```